kpi/views/current_user.py

```diff
@@ -1,13 +1,128 @@
 # coding: utf-8
+from constance import config
 from django.contrib.auth.models import User
-from rest_framework import viewsets
+from django.db import transaction
+from django.utils.timezone import now
+from django.utils.translation import gettext as t
+from rest_framework import status, viewsets
+from rest_framework.response import Response
 
+from kobo.apps.trash_bin.utils import move_to_trash
 from kpi.serializers import CurrentUserSerializer
 
 
 class CurrentUserViewSet(viewsets.ModelViewSet):
+    """
+    <pre class="prettyprint">
+    <b>GET<b> /me/
+    </pre>
+
+    > Example
+    >
+    >       curl -X GET https://[kpi]/me/
+    >
+    >       {
+    >           "username": string,
+    >           "first_name": string,
+    >           "last_name": string,
+    >           "email": string,
+    >           "server_time": "YYYY-MM-DDTHH:MM:SSZ",
+    >           "date_joined": "YYYY-MM-DDTHH:MM:SSZ",
+    >           "projects_url": "https://[kpi]/{username}",
+    >           "is_superuser": boolean,
+    >           "gravatar": url,
+    >           "is_staff": boolean,
+    >           "last_login": "YYYY-MM-DDTHH:MM:SSZ",
+    >           "extra_details": {
+    >               "bio": string,
+    >               "city": string,
+    >               "name": string,
+    >               "gender": string,
+    >               "sector": string,
+    >               "country": string,
+    >               "twitter": string,
+    >               "linkedin": string,
+    >               "instagram": string,
+    >               "organization": string,
+    >               "require_auth": boolean,
+    >               "last_ui_language": string,
+    >               "organization_website": sting,
+    >           },
+    >           "git_rev": {
+    >               "short": boolean,
+    >               "long": boolean,
+    >               "branch": boolean,
+    >               "tag": boolean,
+    >           },
+    >           "social_accounts": [],
+    >       }
+
+    Update account details
+    <pre class="prettyprint">
+    <b>PATCH<b> /me/
+    </pre>
+
+    > Example
+    >
+    >       curl -X PATCH https://[kpi]/me/
+
+    > Payload Example
+    >
+    >       {
+    >           "first_name": "Bob",
+    >           "social_accounts": [],
+    >       }
+
+    Delete the entire account
+    <pre class="prettyprint">
+    <b>DELETE<b> /me/
+    </pre>
+
+    >   Example
+    >
+    >       curl -X DELETE https://[kpi]/me/
+
+    > Payload Example
+    >
+    >       {
+    >           "conifrm": {user__extra_details__uid},
+    >       }
+
+
+    ### Current User Endpoint
+    """
     queryset = User.objects.none()
     serializer_class = CurrentUserSerializer
 
+    # TODO: Fix `social_accounts` so it's not required in PATCH request KPI#4397
```

Comment:
I think this might not be true; see https://github.com/kobotoolbox/kpi/issues/4397#issuecomment-1492476607